mountpoint-s3/src/fs.rs

```diff
@@ -155,6 +155,34 @@ where
 
     pub async fn init(&self, config: &mut KernelConfig) -> Result<(), libc::c_int> {
         let _ = config.add_capabilities(fuser::consts::FUSE_DO_READDIRPLUS);
+        // Set max_background FUSE parameter to 64 by default, or override with environment variable:
+        // NOTE: Support for this environment variable may be removed in future without notice.
+        if let Some(user_max_background) = std::env::var_os("UNSTABLE_MAX_BACKGROUND") {
+            let max_background = user_max_background
+                .to_string_lossy()
+                .parse::<u16>()
+                .expect("invalid env var value for UNSTABLE_MAX_BACKGROUND");
+            let old = config
+                .set_max_background(max_background)
+                .expect("unable to set max background");
+            tracing::warn!("set max background to {} from {}", max_background, old)
+        } else {
+            let _ = config
+                .set_max_background(64)
+                .expect("unable to set max background to default value");
+        }
```

Comment:
Since we're putting this in the `main` branch, I'd rather us handle parsing and setting errors properly and gracefully fail.

As an example, this is a change where we introduced an override for setting the prefetch read window size: https://github.com/awslabs/mountpoint-s3/commit/de6d145c2343b3bce433d1368c71f7eaf2f2d4d7

Reply:
Discussed a bit offline, this is where we're landing.

- If customer is intending to modify an unstable configuration but we don't understand it or can't use it, we want to be loud and tell them its not working. (Crash preferable?)
- If customer sets some unstable configuration, we want to confirm it so they're reassured its applied and can be reminded that an unstable configuration is in use.
- If we remove it in the future, we want to be able to silently take it away without breaking the existing use case (as is the 'contract' with most of these unstable features). In this case, we should silently fail (maybe with a warning for a bit if we feel it is necessary enough).


Given this, keeping the crash here is the right choice (in lieu of plumbing everything through to the CLI error reporting which I don't want to recommend right now).

Reply:
Ack, thanks.  I have still improved the error messages. One thing I hadn't realized initially is that panics here don't actually propagate to the CLI (and this pattern already exists elsewhere), in that the mount will fail but the CLI prints the "success" error message and the metrics thread will continue to run (even with `-f`).  I haven't debugged fully yet though think this could be related to the context in which `init` is called, probably from a fuse thread.

I think we should merge as-is anyway, and then later refactor all similar panics to address this issue, since the unstable flag is experimental, and a failure wouldn't be encountered during normal code paths (setting `max_background` only fails if `0` is passed, which it isn't in the default/safe code path).